.github/workflows/CI.yml

```
@@ -35,27 +37,13 @@ jobs:
         os:
           - ubuntu-latest
     steps:
-      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
-
-      - uses: julia-actions/setup-julia@0b9b1d2cd24245f151902702d8e73b3f6b910014
+      - uses: actions/checkout@v4
```

I'd recommend doing this:
```suggestion
      - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938 # v4.2.0
```

This way, you get the reproducibility of the commit hash, but you still have the human-readable version in the comment.

Dependabot will update both the commit hash and also the comment with the version number.

---

We can keep commit hashes if you want but since we're only using some very well-established actions I'm not worried about malicious actors or semver violations, which IIUC are really the only reasons why one would want to use commit hashes. IMO the tradeoff for brevity/readability here seems worthwhile.

---

I don't feel particular strongly either way. We could do `action@v1.2.3`. That gives us the ability to easily roll back if something breaks, and is less verbose than `action@commithash # v1.2.3`. Dependabot works with any of these formats.